zap/src/test/java/org/parosproxy/paros/core/scanner/VariantUserDefinedUnitTest.java

```diff
@@ -0,0 +1,113 @@
+ * Copyright 2025 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.parosproxy.paros.core.scanner;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+/** Unit test for {@link VariantUserDefined}. */
+class VariantUserDefinedUnitTest {
+
+    @Test
+    void shouldHaveParametersListEmptyByDefault() {
+        // Given
+        VariantUserDefined variant = new VariantUserDefined();
+        // When
+        List<NameValuePair> parameters = variant.getParamList();
+        // Then
+        assertThat(parameters, is(empty()));
+    }
+
+    @Test
+    void shouldFailToExtractParametersFromUndefinedMessage() {
+        // Given
+        VariantUserDefined variant = new VariantUserDefined();
+        HttpMessage undefinedMessage = null;
+        // When / Then
+        assertThrows(IllegalArgumentException.class, () -> variant.setMessage(undefinedMessage));
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "0,29,true", // Entire Header - Silly example, but still
+        "29,29,true", // At end of header
+        "0,0,true", // At start of header
+        "5,8,true", // Within range
+        "33,33,false", // Beyond end
+        "25,33,false" // In to out
+    })
+    void shouldEvaluateHeaderBoundsProperly(int start, int end, boolean expected) {
+        // Given
+        VariantUserDefined variant = new VariantUserDefined();
+        HttpMessage msg = createBasicMessage();
+        // When
+        variant.setMessage(msg);
+        boolean inHeader = variant.isInHeader(new int[] {start, end});
+        // Then
+        assertThat(inHeader, is(equalTo(expected)));
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "0,29,false", // Entire Header - Silly example, but still
+        "29,29,false", // At end of header
+        "0,0,false", // At start of header
+        "5,8,false", // Within header range
+        "33,33,true", // Beyond end of header
+        "25,33,false", // Within header to beyond header
```

Comment:
IMO only one less than the start of the body is necessary to test the header boundary.

Reply:
I've reduced this to a single unit test method. It still includes a bunch of header cases. Let me know if you still want it reduced.